tools/rpm2img

```diff
@@ -254,20 +255,34 @@ veritysetup verify "${ROOT_IMAGE}" "${VERITY_IMAGE}" "${VERITY_ROOT_HASH}"
 dd if="${VERITY_IMAGE}" of="${OS_IMAGE}" conv=notrunc bs=1M seek="${partoff[HASH-A]}"
 
 # write GRUB config
+GRUB_FEATURES_ARRAY=("${GRUB_FEATURES}")
+# If GRUB_SET_PRIVATE_VAR is set, include the parameters that support Boot Config
+if [[ "${GRUB_FEATURES_ARRAY[*]}" =~ "GRUB_SET_PRIVATE_VAR" ]]; then
+   BOOTCONFIG='bootconfig'
+   INITRD='initrd ($private)/bootconfig.data'
+else
```

Comment:
nit: might be better to test each element specifically, in the unlikely event that one is a substring of another:
```suggestion
# If GRUB_SET_PRIVATE_VAR is set, include the parameters that support Boot Config
if printf '%s\n' "${GRUB_FEATURES_ARRAY[@]}" | grep -Fxq 'GRUB_SET_PRIVATE_VAR' ; then
   BOOTCONFIG='bootconfig'
   INITRD='initrd ($private)/bootconfig.data'
else
```